lib/metrics/metric_api.rb

```diff
@@ -112,7 +112,7 @@ def push_data(id, data)
       def get_data(id, starts: nil, ends: nil)
         params = {:start => starts, :end => ends}
         params.delete_if { |k, v| v.nil? }
-        resp = @client.http_get("/#{@resource}/#{id}/data/?"+URI.encode_www_form(params))
+        resp = @client.http_get("/#{URI.escape(@resource)}/#{id}/data/?"+URI.encode_www_form(params))
```

Comment:
We have a good deal of strange chars in metric ids, so that escaping looks right to me.